cmake/Monolithic.cmake

```diff
@@ -19,4 +19,5 @@ if(LY_MONOLITHIC_GAME)
 else()
     ly_set(PAL_TRAIT_MONOLITHIC_DRIVEN_LIBRARY_TYPE SHARED)
     ly_set(PAL_TRAIT_MONOLITHIC_DRIVEN_MODULE_TYPE GEM_MODULE)
+    ly_set(PAL_TRAIT_MONOLITHIC_DRIVEN_GEM_SHARED_TYPE GEM_SHARED)
```

Comment:
The top half of the if block needs to be added as well
That should be `ly_set(PAL_TRAIT_MONOLITHIC_DRIVEN_GEM_SHARED_TYPE GEM_STATIC)`